src/metadata.rs

```diff
@@ -21,8 +36,14 @@ pub fn build_metadata<T>(
 where
     T: Meta,
 {
+    let mut merged_labels = get_recommended_labels(resource)?;
+
+    if let Some(provided_labels) = labels {
+        merged_labels.extend(provided_labels);
```

Comment:
This would overwrite existing ones?

If so: Could you add a comment to the docs about that?